hasher-matcher-actioner/terraform/variables.tf

```diff
@@ -52,14 +52,14 @@ variable "te_api_token" {
   sensitive   = true
 }
 
-variable "collab_file" {
-  description = "An optional file name of ThreatExchange Collaborations objects to prepopulate. See collabs_example.json for the correct formatting"
-  type = string
-  default = "collabs_example.json"
-}
-
 variable "fetch_frequency" {
   description = "How long to wait between calls to ThreatExcahnge. Must be an AWS Rate Expression. See here: https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html"
   type        = string
   default     = "15 minutes"
 }
+
+variable "collab_file" {
```

Comment:
(Question that I didn't ask on the original diff): How do optional values work? It looks like its used unconditionally, where does the optional part come in?

Reply:
I wasn't sure (and honestly didn't look into) whether terraform supported optional variables. The script first checks if the file given exists and will only populate aws if it does. Thus the optional part come in by having a default value for a file that doesn't exists.

I'm realizing now that having a default value that looks like a file is probably confusing. Perhaps I should change the description to "A file name of ThreatExchange Collaborations objects. If the file exists, the configs in it will be pre-populated into AWS. See collabs_example.json for the correct formatting"

Reply:
A pattern I used once was to have a bool to trigger processing, and optionally required variables used by the processing gated on the bool. See the variable `use_shared_user_pool` and the related variables  `webapp_and_api_shared_user_pool_id` and `webapp_and_api_shared_user_pool_client_id`.